dd-java-agent/instrumentation/javax-mail/src/main/java/datadog/trace/instrumentation/javax/mail/JavaxMailInstrumentation.java

```diff
@@ -0,0 +1,78 @@
+package datadog.trace.instrumentation.javax.mail;
+
+import static datadog.trace.agent.tooling.bytebuddy.matcher.NameMatchers.named;
+
+import com.google.auto.service.AutoService;
+import datadog.trace.agent.tooling.Instrumenter;
+import datadog.trace.agent.tooling.InstrumenterModule;
+import datadog.trace.api.iast.InstrumentationBridge;
+import datadog.trace.api.iast.Sink;
+import datadog.trace.api.iast.VulnerabilityTypes;
+import datadog.trace.api.iast.sink.EmailInjectionModule;
+import java.io.IOException;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Part;
+import javax.mail.internet.MimeMultipart;
+import net.bytebuddy.asm.Advice;
+
+@AutoService(InstrumenterModule.class)
+public class JavaxMailInstrumentation extends InstrumenterModule.Iast
+    implements Instrumenter.ForSingleType, Instrumenter.HasMethodAdvice {
+
+  public JavaxMailInstrumentation(String instrumentationName, String... additionalNames) {
+    super("javaxmailinstrumentation", "javaxmail");
+  }
+
+  @Override
+  public void methodAdvice(MethodTransformer transformer) {
+    transformer.applyAdvice(
+        named("javaxMailInstrumentation"),
```

Comment:
```suggestion
        named("send"),
```
The first argumente of the applyAdvice is the element matcher and the second the Advice. In our case we need to instrument the method `send`

it's also important to note that this will instrument all methods named send within the class, applying the advice every time the method is called. This could be an issue because, if we have overloaded methods (as in our case) and some of them internally call others, the Advice logic will be applied multiple times.

There are several options to address this, but the simplest and most straightforward approach is to analyze how these methods behave internally. Ideally, all methods would eventually call a single one that is responsible for actually sending the email. If that’s the case, filters based on the parameters should be added in the applyAdvice logic to ensure it is applied only where necessary.

Could you check how they are implemented? 😄

Reply:
so in this case, the overloaded methods do not call each other but a private method, send0, i will show you when we have our meeting